test/instrumentatios/extentions/amqplib.test.ts

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright The Cisco Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// in order to these tests locally, run: docker run `export RUN_RABBITMQ_TESTS=1'; npm run rabbitmq`
+import { AmqplibInstrumentation } from 'opentelemetry-instrumentation-amqplib';
+import {
+  BasicTracerProvider,
+  InMemorySpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import * as assert from 'assert';
+
+const instrumentation = new AmqplibInstrumentation();
+instrumentation.enable();
+const memoryExporter = new InMemorySpanExporter();
+const provider = new BasicTracerProvider();
+instrumentation.setTracerProvider(provider);
+provider.addSpanProcessor(new SimpleSpanProcessor(memoryExporter));
+
+import * as amqp from 'amqplib';
+import * as amqpCallback from 'amqplib/callback_api';
```

Comment:
I suggest using import { Channel, ConfirmChannel} from 'amqlib/callback_api' to keep file a bit smaller